nilearn/plotting/displays.py

```diff
@@ -1216,6 +1221,9 @@ def add_graph(self, adjacency_matrix, node_coords,
                               vmin=edge_vmin, vmax=edge_vmax,
                               **edge_kwargs)
 
+        if colorbar:
+            self._colorbar_show(ax, threshold=edge_threshold)
```

Comment:
> I don't see anything nasty here... He is using a method of its own class, this is totally legit.

The `im` argument in _colorbar_show is expected to be some kind of image object as produced by plt.imshow (matplotlib.image.AxesImage), hence the name. At least this is what happening in the rest of the code. Suddenly you are passing some kind of axis object which is very surprising. "How can that even work" is how I felt until I realised that the .cmap and .norm attributes were set on the GlassBrainAxes instance just for the sole purpose of making _colorbar_show work.

> The question is whether we accept to have this hack until somebody decide to clean this code.

I would say yes if we want this to be in the release. Clear no, otherwise, but that just my feeling.

I don't think it is that much work to do what I proposed. There may be better alternatives too.